source/Ship.cpp

```diff
@@ -3403,7 +3425,77 @@ double Ship::MaxReverseVelocity() const
 // DamageDealt from that weapon. The return value is a ShipEvent type,
 // which may be a combination of PROVOKED, DISABLED, and DESTROYED.
 // Create any target effects as sparks.
-int Ship::TakeDamage(vector<Visual> &visuals, const DamageDealt &damage, const Government *sourceGovernment)
+int Ship::TakeDamage(vector<Visual> &visuals, const DamageDealt &damage, const Government *sourceGovernment,
+	Point intersection, Point hitVelocity, Angle hitAngle, bool splash)
+{
+	int type = DoDamage(visuals, damage);
+
+	// If this ship did not consider itself an enemy of the ship that hit it,
+	// it is now "provoked" against that government.
+	if(sourceGovernment && !sourceGovernment->IsEnemy(government)
+			&& (Shields() < .9 || Hull() < .9 || !personality.IsForbearing())
+			&& !personality.IsPacifist() && damage.GetWeapon().DoesDamage())
+		type |= ShipEvent::PROVOKE;
```

Comment:
I can move it back into DoDamage if that is required, though I'd need to redo the arguments into it. I initially moved it because the actual application of damage doesn't require that bit itself.

What is currently in DoDamage is really the only shared part of the damage pathway for weather and weapons damage. As this is only something to do with weapons damage, I'd say personally I'd keep it with the handler for weapons damage rather than the actual applicator of any and all damage.

As I said, I'm happy to move it back to DoDamage, but I personally would first go to leave it outside of DoDamage. I'll see what you think to my reasoning. If it's still to put it in DoDamage I can do that.